cockatrice/src/cardinfowidget.cpp

```diff
@@ -153,17 +153,29 @@ void CardInfoWidget::setCard(CardInfo *card)
     if (info)
         disconnect(info, 0, this, 0);
     info = card;
-    connect(info, SIGNAL(pixmapUpdated()), this, SLOT(updatePixmap()));
-    connect(info, SIGNAL(destroyed()), this, SLOT(clear()));
+    if(info)
+    {
+        connect(info, SIGNAL(pixmapUpdated()), this, SLOT(updatePixmap()));
+        connect(info, SIGNAL(destroyed()), this, SLOT(clear()));
+
+        nameLabel2->setText(info->getName());
+        manacostLabel2->setText(info->getManaCost());
+        colorLabel2->setText(info->getColors().join(""));
+        cardtypeLabel2->setText(info->getCardType());
+        powtoughLabel2->setText(info->getPowTough());
+        loyaltyLabel2->setText(info->getLoyalty() > 0 ? QString::number(info->getLoyalty()) : QString());
+        textLabel->setText(info->getText());
+    } else {
+        nameLabel2->setText("");
+        manacostLabel2->setText("");
+        colorLabel2->setText("");
+        cardtypeLabel2->setText("");
+        powtoughLabel2->setText("");
+        loyaltyLabel2->setText("");
+        textLabel->setText("");
```

Comment:
Seems like these labels are somewhat duplicated from CardInfoText; maybe there should be some shared behavior between them to reduce it.